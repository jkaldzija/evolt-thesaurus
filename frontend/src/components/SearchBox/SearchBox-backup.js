import React from "react";
import {Button, Card, CardContent, TextField, Typography} from "@material-ui/core";
import './style.css';

type SearchBoxProps = {
    onSubmit: (string) => void
}

export default class SearchBox extends React.Component<SearchBoxProps>{
    state = {
        word: ""
    };

    onSubmit = (e) => {
        e.preventDefault();
        this.props.onSubmit(this.state.word);
    };

    handleChange = e => this.setState({word: e.target.value});

    render() {
        const {word} = this.state;
        return (
            <Card className="search-container">
                <CardContent>
                    <Typography variant="h6" color="default" className="search-title">
                        Search for synonyms
                    </Typography>
                    <form onSubmit={this.onSubmit}>
                        <TextField value={word}
                                   placeholder="Enter word"
                                   fullWidth={true}
                                   onChange={this.handleChange}/>
                        <div className="search-button-container">
                            <Button type="submit" variant="contained" color="primary">Search</Button>
                        </div>
                    </form>
                </CardContent>
            </Card>
        );
    }
}
