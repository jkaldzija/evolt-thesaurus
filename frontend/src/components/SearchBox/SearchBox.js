import React from 'react';
import {Card, Container, Input, Spinner} from "reactstrap";
import style from "./searchBox.module.css";
import classnames from "classnames";
import ThemeButton from "../Form/ThemeButton/ThemeButton";
import {WithLoading} from "../HOC/WithLoading";
import SynonymList from "../SynonymList/SynonymList";
import {withRouter} from "react-router";
import * as qs from "qs";
import wordService from "../../api/wordService";
import {getSafe, getSafeDeep} from "../Utility/state";
import {getQueryParam} from "../Utility/routing";

const SynonymListWithLoading = WithLoading(SynonymList);

class SearchBox extends React.Component {
    constructor(props) {
        super(props);

        let currentSearch = getQueryParam(this.props.location.search, "q");
        if(!currentSearch) currentSearch = "";

        this.state = {
            search: currentSearch,
            currentWord: undefined,
            isLoading: false
        }
    }

    performSearch = async() => {
        if(this.state.search === "") return;
        this.setState({
            isLoading: true
        } );
        let response = {};
        try{
            response = await wordService.getSynonymsForWord(getSafe(this.state.search));
        }catch (e) {

        }
        this.setState({
            currentWord: getSafe(response.data),
            isLoading: false
        });

    };

    componentDidMount(){
        this.performSearch();
    }

    componentDidUpdate(prevProps, prevState, SS){
        const prevSearch = getQueryParam(prevProps.location.search, "q");
        const search = getQueryParam(this.props.location.search, "q");
        console.log(prevSearch + "   " + search);
        if(search != prevSearch){
            this.setState({ search }, () => {
                this.performSearch();
            });

        }

    }


    onSearch = (e) => {
        e.preventDefault();
        let params = qs.parse(this.props.pathname);
        params.q = this.state.search;

        this.props.history.push(`?${qs.stringify(params)}`);

    };


    onInput = e => {
        this.setState({
            [e.target.name]: e.target.value
        })
    };

    reloadData = () => {
        this.performSearch();
    };

    render() {
        return  <Card>
                    <form onSubmit={this.onSearch} className={classnames("d-flex", style.form)}>

                        <Input placeholder={"Please enter a word to find synonims"}
                            className={classnames(style["search-input"])}
                            value={this.state.search}
                            name={"search"}
                            onChange={this.onInput}
                        />
                        <ThemeButton className={style.button}>
                            Search
                        </ThemeButton>

                    </form>
                    <SynonymListWithLoading
                        reloadData={this.reloadData}
                        currentWord={this.state.currentWord}
                        isLoading={this.state.isLoading}
                        synonymList={getSafeDeep(this.state, "currentWord.synonyms", [])}  />


                </Card>
    };

}

export default withRouter(SearchBox);
