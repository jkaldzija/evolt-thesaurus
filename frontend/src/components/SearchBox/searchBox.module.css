

.search-container {
    padding: 24px;
}

.search-title {
    margin-bottom: 12px !important;
}

.button{
    margin-left: 40px;
}

.form{
    margin-bottom:35px;
}

.search-button-container {
    margin-top: 24px;
    display: flex;
    justify-content: center;
}

.search-input{
    border-top: 0px;
    border-left: 0px;
    border-right: 0px;
    border-radius: 0;
}
