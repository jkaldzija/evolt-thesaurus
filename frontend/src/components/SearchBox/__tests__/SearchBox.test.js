import React from 'react';
import {shallow} from "enzyme";
import SearchBox from "../SearchBox";

describe('Search Box', function () {
    it('should render correctly', function () {
        const component = shallow(
            <SearchBox onSubmit={() => {}}/>
        );
        expect(component).toMatchSnapshot();
    });

    it('should handle input change', function () {
        const component = shallow(
            <SearchBox onSubmit={() => {}}/>
        );
        component.instance().handleChange({target: {value: "test"}});
        expect(component.state("word")).toBe("test")
    });

    it('should send submit event', function () {
        const submitCallback = jest.fn();
        const component = shallow(
            <SearchBox onSubmit={submitCallback}/>
        );

        component.instance().setState({word: "test"});
        component.instance().onSubmit({preventDefault: () => {}});
        expect(submitCallback).toHaveBeenCalledWith("test");
    });
});
