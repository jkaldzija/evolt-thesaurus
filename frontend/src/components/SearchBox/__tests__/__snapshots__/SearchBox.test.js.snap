// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Box should render correctly 1`] = `
<WithStyles(Card)
  className="search-container"
>
  <WithStyles(CardContent)>
    <WithStyles(Typography)
      className="search-title"
      color="default"
      variant="h6"
    >
      Search for synonyms
    </WithStyles(Typography)>
    <form
      onSubmit={[Function]}
    >
      <TextField
        fullWidth={true}
        onChange={[Function]}
        placeholder="Enter word"
        required={false}
        select={false}
        value=""
        variant="standard"
      />
      <div
        className="search-button-container"
      >
        <WithStyles(Button)
          color="primary"
          type="submit"
          variant="contained"
        >
          Search
        </WithStyles(Button)>
      </div>
    </form>
  </WithStyles(CardContent)>
</WithStyles(Card)>
`;
